### PR TITLE
director transmits its own extra hosts to spawned services *demo

### PR DESCRIPTION
**Pull request by sanderegg**
if there are extra hosts defined for the director it will pass them to the services

## What do these changes do?

- [x] reads in the system extra hosts
- [x] passes the system extra hosts to the spawned services

## Related issue number
connects #305 

## Checklist

- [x] I think the code is well written
- [ ] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] **Runs in the swarm**
- [x] If you design a new module, add your user to .github/CODEOWNERS

[waffle.io]:https://waffle.io/marketing-assets/documents/waffleio_cheatsheet_v1.pdf?utm_source=blog&utm_medium=cheatsheet-ctabutton&utm_campaign=cheatsheet

### CODE REVIEW

File: services/director/src/simcore_service_director/system_utils.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from pathlib import Path
+from typing import Dict
+
+
+def get_system_extra_hosts(extra_host_domain: str) -> Dict:
+    extra_hosts = {}
+    hosts_path = Path("/etc/hosts")
+    if hosts_path.exists():
```
Comment (pcrespov) [inline services/director/src/simcore_service_director/system_utils.py]: `` if hosts_path.exists() and extra_host_domain!="undefined": ``

File: services/director/tests/test_producer.py [hunk 2]
```diff
@@ -90,3 +90,8 @@
     service_information = low_level_client.inspect_service(docker_service.id)
     service_published_port = service_information["Endpoint"]["Ports"][0]["PublishedPort"]
     assert service_published_port == service_port
+
+async def test_extra_hosts_passed_to_services(run_services):
+    # would need to test right inside a docker or test from outside...
+    # start the director with extra hosts, start some services, and test if the extra hosts are added
+    pass
```
Comment (pcrespov) [inline services/director/tests/test_producer.py]: Does this test ``pass``-es  ;-)

Comment (sanderegg) [reply services/director/tests/test_producer.py]: very nicely indeed!

File: services/director/src/simcore_service_director/system_utils.py (hunk 1 above)
Comment (sanderegg) [reply services/director/src/simcore_service_director/system_utils.py]: done

**Pull request merged**